src/wwwroot/app/directives/event-calendar/event-calendar.js

```diff
@@ -29,63 +33,106 @@ export default class EventCalendarDirective {
+   vm.goToPreviousMonth         = goToPreviousMonth;
+   vm.goToNextYear              = goToNextYear;
+   vm.goToPreviousYear          = goToPreviousYear;
+   vm.removeCurrentEvent        = removeCurrentEvent;
+   vm.saveCurrentEvent          = saveCurrentEvent;
+   vm.onDblclick                = showAddDialog;
+   vm.checkedUsers              = [];
+   vm.eventsForDay              = [];
+   vm.getEventsForDate          = getEventsForDate;
+   vm.getDailyEvents            = getDailyEvents;
+   vm.numberEventsForDailyView  = 3;
+   vm.lengthOfWeek              = 7;
+   vm.daysOfWeek                = ['Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat', 'Sun'];
+
+   (function init() {
+      formatingSearchCondition();
+      getDaysInMonth(vm.startDate, vm.endDate);
+      vm.getEvents(vm.startDate, vm.endDate, vm.checkedUsers).then(convertEventsToHash);
+   }());
 
    function showAddDialog(day) {
       vm.$broadcast('onDblclick', {date: day});
    }
 
+   function formatingSearchCondition() {
+      vm.startDate = new Date(vm.currentYear, vm.currentMonth, 1);
+      vm.firstDayOfWeek = vm.startDate.getDay();
+      if (vm.firstDayOfWeek === 0) {
+         vm.startDate.setDate(vm.startDate.getDate() - (vm.lengthOfWeek - 1));
+      } else {
+         vm.startDate.setDate(vm.startDate.getDate() - (vm.firstDayOfWeek - 1));
+      }
+      vm.endDate = new Date(vm.dateForSearchEvents.getFullYear(), vm.dateForSearchEvents.getMonth() + 1, 0);
+      vm.lastDayOfWeek = vm.endDate.getDay();
+      if (vm.lastDayOfWeek !== 0) {
+         vm.endDate.setDate(vm.endDate.getDate() + (vm.lengthOfWeek - vm.lastDayOfWeek));
+      }
+   }
+
    vm.$on('checkUser', function fromParent(event, obj) {
-      vm.checkedUsers = obj.chekedUsers;
-      vm.eventsForMonth = {};
-      vm.getEvents(vm.dateForSearchEvents, vm.chekedUsers).then((events) => {
-         convertEventsToHash(events);
-      });
+      let diff = [];
+      let clonedCheckedUsersIds = clone(obj.checkedUsersIds);
+      formatingSearchCondition();
+      if (vm.checkedUsers.length === 0) {
+         vm.eventsForMonth = {};
+         vm.getEvents(vm.startDate, vm.endDate, obj.checkedUsersIds).then((events) => {
+            convertEventsToHash(events);
+            getEventsForDate();
+         });
+      } else if (vm.checkedUsers.length > obj.checkedUsersIds.length) {
+         let unCheckedUserId = difference(vm.checkedUsers, obj.checkedUsersIds);
+         mapKeys(vm.eventsForMonth, value => {
+            remove(value, {responsibleId: unCheckedUserId[0].toString()});
+         });
+      } else {
+         diff = difference(obj.checkedUsersIds, vm.checkedUsers);
+         vm.getEvents(vm.startDate, vm.endDate, diff).then((events) => {
+            convertEventsToHash(events);
+            getEventsForDate();
+         });
+      }
+      vm.checkedUsers = clonedCheckedUsersIds;
    });
 
-   function getDaysInMonth(month, year) {
+   function getDaysInMonth(startDt, endDt) {
```

Comment:
this method return nothing as I see, so it can't be get-method. As well as your method not only set days in month, but also set current-date and execute other functions.  So you can make this method more independence and cut into several smaller functions or rename it.